src/js_app.hpp

```diff
@@ -30,14 +30,34 @@
 #include "js_network_transport.hpp"
 #include "js_email_password_auth.hpp"
 
+#include <forward_list>
+
 using SharedApp = std::shared_ptr<realm::app::App>;
 using SharedUser = std::shared_ptr<realm::SyncUser>;
+using AppToken = realm::Subscribable<realm::app::App>::Token;
 
 namespace realm {
 namespace js {
 
 template <typename T>
-class AppClass : public ClassDefinition<T, SharedApp> {
+class App {
+public:
+    App(const SharedApp& l)
+        : m_app(l)
+    {
+    }
+
+    App(const App&) = delete;
```

Comment:
I think the comment from `User` is a good one to have here too
```suggestion
    // Remove copy constructors to avoid destroying the listener Token
    App(const App&) = delete;
```